### PR TITLE
Add mysql replica APIs into spec

### PR DESCRIPTION
**Pull request by Suna-MS**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by anuchandy** [comment]
@Suna-MS are you planning to add more changes for replica? (i see api to create and list replicas). Or this is the final version?

**Comment by anuchandy** [comment]
@Suna-MS just to clarify, the reason I asked about completion status is to see whether it's ready for ARM review so that we can move faster.

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/stable/2017-12-01/examples/ReplicasListByServer.json [hunk 1]
```diff
@@ -0,0 +1,109 @@
+{
+  "parameters": {
+    "serverName": "testprimary",
+    "resourceGroupName": "TestGroup",
+    "api-version": "2017-12-01-preview",
```
Comment (QingqingYuan) [inline specification/mysql/resource-manager/Microsoft.DBforMySQL/stable/2017-12-01/examples/ReplicasListByServer.json]: this is under stable folder. Please use the stable version as well

**Review by anuchandy — APPROVED** [review]

**Review by QingqingYuan — APPROVED** [review]

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/stable/2017-12-01/mysql.json [hunk 2]
```diff
@@ -283,6 +286,45 @@
...
+          },
+          {
+            "$ref": "#/parameters/ServerNameParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/ServerListResult"
```
[the remark this answers is not in the document]
Comment (Suna-MS) [reply specification/mysql/resource-manager/Microsoft.DBforMySQL/stable/2017-12-01/mysql.json]: Replica server is also a server, not a new resource type. Replica - Primary is a relation between two servers.  This API is to get all replica servers that belongs to a specified (primary) server, not all replica servers in a resource group.

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/stable/2017-12-01/mysql.json [hunk 3]
```diff
@@ -283,6 +286,45 @@
         }
       }
     },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DBforMySQL/servers/{serverName}/Replicas": {
```
[the remark this answers is not in the document]
Comment (Suna-MS) [reply specification/mysql/resource-manager/Microsoft.DBforMySQL/stable/2017-12-01/mysql.json]: fixed

**Comment by Suna-MS** [comment]
The backend changes is rolling out to production this week, will update the spec accordingly once the word-wide deployment finished. The only change needed is update keyword from 'Primary' to 'Master'.

**Comment by anuchandy** [comment]
@Suna-MS just checking - is this deployed?

**Comment by Suna-MS** [comment]
The public review day postponed again, waiting for new schedule.

**Comment by Suna-MS** [comment]
@anuchandy, we can go ahead for this PR and if no other comments, can check it in.

**Comment by Suna-MS** [comment]
@veronicagg @anuchandy, we can continue this PR currently, can you please help to merge it?

**Comment by veronicagg** [comment]
@Suna-MS sure, @anuchandy who did the review can take it from here. thanks!

**Comment by Suna-MS** [comment]
@anuchandy, any updates?

**Comment by anuchandy** [comment]
Merging this PR. There is a example validation error, but doesn't look like issue with this PR. Opened an issue here https://github.com/Azure/oav/issues/325 to triage.

**Review by anuchandy — APPROVED** [review]

**Pull request merged**